security/src/main/java/org/aggregatortech/dindora/security/authentication/interceptor/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,38 @@
+package org.aggregatortech.dindora.security.authentication.interceptor;
+
+import org.aggregatortech.dindora.common.ServiceLocatorHelper;
+import org.aggregatortech.dindora.exceptions.IDStoreNotConfiguredException;
+import org.aggregatortech.dindora.exceptions.InvalidCredentialsException;
+import org.aggregatortech.dindora.exceptions.MessageService;
+import org.aggregatortech.dindora.security.authentication.providers.IDProvider;
+import org.aggregatortech.dindora.security.authentication.token.AuthenticationCredentials;
+
+import java.io.IOException;
+
+public class AuthenticationInterceptor {
+
+
+    private final IDProvider provider;
+
+    public AuthenticationInterceptor() throws IDStoreNotConfiguredException {
+
+        provider = ServiceLocatorHelper.getServiceLocator().getService(IDProvider.class);
+
+
+        provider.configure();
```

Comment:
Why do we need to invoke the configure() method from outside? Can we have the provider configure itself?

Reply:
pls see above about contract and this class implementing interface

Reply:
Actually the question is about the contract itself i.e. I am trying to understand the need for separate public method to configure the provider. Looks like it should be possible for the provider to configure itself when its instantiated.